rules/no-useless-promise-resolve-reject.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+const {methodCallSelector} = require('./selectors/index.js');
+
+const RETURN_RESOLVE = 'return-resolve';
+const RETURN_REJECT = 'return-reject';
+const YIELD_RESOLVE = 'yield-resolve';
+const YIELD_REJECT = 'yield-reject';
+const messages = {
+	[RETURN_RESOLVE]: 'Prefer `return value` over `return Promise.resolve(value)`.',
+	[RETURN_REJECT]: 'Prefer `throw value` over `return Promise.reject(value)`.',
```

Comment:
```suggestion
	[RETURN_REJECT]: 'Prefer `throw error` over `return Promise.reject(error)`.',
```

I know you can technically throw/reject any value, but the common case is an error, so we  should optimize the message for that.